### PR TITLE
Router

### PR DESCRIPTION
**Pull request by BJordi**
## Summary

- Renamed Login component to SignUp.
- Added Login component.
- Added routes handling.

## Screenshots

02-routes
03-routes

### CODE REVIEW

File: src/router.js [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
 
 Vue.use(Router)
 
 export default new Router({
   routes: [
     {
-      path: '/',
+      path: '/sign-up',
+      name: 'signUp',
+      component: SignUp
```
Comment (FrankIglesias) [inline src/router.js]: use the `() => import(....)` way

File: src/views/Login.vue [hunk 2]
```diff
@@ -96,25 +61,8 @@ export default {
...
-      }
-      return formObject
-    },
     onSubmit () {
-      this.submitted = true
-      if (!this.$v.$invalid) {
-        const userData = this.generateObject()
-        signUp(userData)
-      }
+      console.log("You're in!")
```
Comment (FrankIglesias) [inline src/views/Login.vue]: remove this comment

File: src/views/SignUp.vue [hunk 3]
```diff
@@ -0,0 +1,145 @@
+<template>
+  <div class="container">
```
Comment (FrankIglesias) [inline src/views/SignUp.vue:2]: transform this into a pug file

File: src/views/SignUp.vue [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    async onSubmit () {
+      this.submitted = true
+      if (!this.$v.$invalid) {
+        const userData = this.generateObject()
+        const response = await signUp(userData)
+        if (response.ok) {
+          this.$router.push('/login')
+        } else {
+          this.formError =
+            response.data && response.data.error && response.data.error[0]
```
Comment (FrankIglesias) [inline src/views/SignUp.vue:107]: why are you checking for the first value of error

File: src/views/SignUp.vue [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    }
+  }
+}
+</script>
+
+<style lang="scss">
+@import "../scss/commons/colors.scss";
+.container {
+  background-color: $light-grey;
+  border: 1px solid $light-grey2;
```
Comment (FrankIglesias) [inline src/views/SignUp.vue:119]: you can use a shorthand for this

File: src/views/SignUp.vue [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+  border-top: 5px solid $scooter;
+  display: flex;
+  flex-direction: column;
+  margin: 100px;
+  padding: 0 15px;
+  width: 300px;
```
Comment (FrankIglesias) [inline src/views/SignUp.vue]: use width and max width

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

File: src/views/SignUp.vue (hunk 3 above)
Comment (BJordi) [reply src/views/SignUp.vue:2]: Will do it in a specific PR.

File: src/views/SignUp.vue (hunk 4 above)
Comment (BJordi) [reply src/views/SignUp.vue:107]: Do you think I should use a map method?

File: src/views/SignUp.vue (hunk 5 above)
Comment (BJordi) [reply src/views/SignUp.vue:119]: Which one? Border is the shorthand to `border-width`, `border-style` and `border-color`? Or are you talking about something else? :thinking:

File: src/router.js (hunk 1 above)
Comment (BJordi) [reply src/router.js]: Is this lazy loading?

Comment (BJordi) [reply src/router.js]: Is it lazy loading?

File: src/router.js [hunk 7]
```diff
@@ -1,15 +1,22 @@
...
+      component: () => import('./views/SignUp.vue')
+    },
+    {
+      path: '/login',
       name: 'login',
-      component: Login
+      component: () => import('./views/Login.vue')
+    },
+    {
+      path: '/', redirect: '/sign-up'
```
Comment (FrankIglesias) [inline src/router.js]: move this  to the below line

File: src/views/SignUp.vue (hunk 5 above)
Comment (FrankIglesias) [reply src/views/SignUp.vue:119]:
```
border: 1px solid;
border-color: $light-grey $light-grey $light-grey $light-grey2;
```

Comment (FrankIglesias) [reply src/views/SignUp.vue:119]: or sort of

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

**Review by FrankIglesias — APPROVED** [review]

**Pull request merged**